cogboard-webapp/src/reducers/widgets/widgetsById.js

```diff
@@ -28,17 +28,19 @@ const receiveData = (state, action) => {
 const updateWidget = (state, { payload }) => {
   const { id, content, ...other } = payload;
   const widget = state[id];
-  const prevContent = widget.content;
+  const prevContent = widget && widget.content;
 
-  return {
-    ...state,
-    [id]: {
-      ...widget,
-      ...other,
-      content: { ...prevContent, ...content },
-      isUpdating: false
-    }
-  };
+  return widget
```

Comment:
I am thinking if it would be a better idea to handle this logic in a thunk because it is not directly related to state shape but a condition of updating or not, so we could check widget existing in a thunk and dispatch action or not.

Reply:
Good point, widget check moved to thunks